File: data/examples/divergent-change.cpp
class Employee {
private:
    std::string name;
    double salary;
    std::string department;
    std::string address;
    std::string email;
    
public:
    // 因为薪资政策变化需要修改
    double calculatePay() {
        if (department == "Engineering") {
            return salary * 1.2;
        } else if (department == "Sales") {
            return salary * 1.1;
        } else if (department == "Management") {
            return salary * 1.5;
        }
        return salary;
    }
    
    // 因为税法变化需要修改
    double calculateTax() {
        double pay = calculatePay();
        if (pay > 100000) {
            return pay * 0.35;
        } else if (pay > 50000) {
            return pay * 0.25;
        }
        return pay * 0.15;
    }
    
    // 因为报告格式变化需要修改
    std::string generateReport() {
        return "Employee: " + name + 
               "\nDepartment: " + department +
               "\nPay: " + std::to_string(calculatePay()) +
               "\nTax: " + std::to_string(calculateTax());
    }
    
    // 因为数据库结构变化需要修改
    void saveToDatabase() {
        // 数据库保存逻辑
        std::cout << "Saving to database..." << std::endl;
    }
    
    // 因为邮件系统变化需要修改
    void sendPayslip() {
        std::string email = "employee@example.com";
        std::cout << "Sending payslip to: " << email << std::endl;
    }
};

File: data/examples/divergent-change.java
class Employee {
    private String name;
    private double salary;
    private String department;
    private String address;
    private String email;
    
    // Needs to be modified due to payroll policy changes
    public double calculatePay() {
        if ("Engineering".equals(department)) {
            return salary * 1.2;
        } else if ("Sales".equals(department)) {
            return salary * 1.1;
        } else if ("Management".equals(department)) {
            return salary * 1.5;
        }
        return salary;
    }
    
    // Needs to be modified due to tax law changes
    public double calculateTax() {
        double pay = calculatePay();
        if (pay > 100000) {
            return pay * 0.35;
        } else if (pay > 50000) {
            return pay * 0.25;
        }
        return pay * 0.15;
    }
    
    // Needs to be modified due to report format changes
    public String generateReport() {
        return "Employee: " + name + 
               "\nDepartment: " + department +
               "\nPay: " + String.valueOf(calculatePay()) +
               "\nTax: " + String.valueOf(calculateTax());
    }
    
    // Needs to be modified due to database structure changes
    public void saveToDatabase() {
        // Database save logic
        System.out.println("Saving to database...");
    }
    
    // Needs to be modified due to email system changes
    public void sendPayslip() {
        System.out.println("Sending payslip to: " + this.email);
    }
    
    // Getters and setters - may change due to data format requirements
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    
    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }
    
    public String getAddress() { return address; }
    public void setAddress(String address) { this.address = address; }
}
